### ISSUE TITLE
Timers not work

### ISSUE THREAD

**Issue by bootchk** [cloudcity/ZeroToBLE-Part2-Swift#3]
The pauseScan and resumeScan timers never fire.  At least they never print.  Xcode   7.3.1, iOS 9.3.5

**Comment by jsoncylu** [comment]
They are supposed to be scheduledTimers, at least as of Swift 3. I would replace the Timer lines to match this:

`_ = Timer.scheduledTimer(timeInterval: timerScanInterval, target: self, selector: #selector(pauseScan), userInfo: nil, repeats: false)
            centralManager.scanForPeripherals(withServices: nil, options: nil)`

**Comment by bootchk** [comment]
I am no expert in Swift but ... I assigned the result to a global "timer" of type NSTimer.  All I can say is: it worked for me.  The construct " _ = " confuses me with regard to whether an object was created and who owns it and when it might be garbage collected.  Especially with regard to a timer object:  I imagine it means that that timer will exist, owned by the OS, will fire and then go out of existence?  But again, seems a little unnatural to me.

**Comment by jsoncylu** [comment]
Yes, it is odd to see it assigned to a "_". With Swift timers, one should call `.invalidate()`to remove it from the run loop.

Edit: actually since I have made it a "scheduledTimer" that doesn't repeat, it automatically deallocates itself once it is fired.
